scripts/main.js

```diff
@@ -0,0 +1,348 @@
+          }
+        } else if (newY !== y) {
+          board[newY][x] = board[y][x];
+          board[y][x] = 0;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-down--${moveDistance}`);
+          }
+        }
+      }
+    }
+  }
+
+  return hasChanged;
+}
+
+function moveLeft() {
+  const originalBoard = board.map((row) => row.slice());
+  let hasChanged = false;
+
+  for (let y = 0; y < 4; y++) {
+    let merged = [false, false, false, false];
+
+    for (let x = 1; x < 4; x++) {
+      if (board[y][x] !== 0) {
+        let newX = x;
+        let moveDistance = 0;
+
+        while (newX > 0 && board[y][newX - 1] === 0) {
+          newX--;
+          moveDistance++;
+        }
+
+        if (
+          newX > 0 &&
+          board[y][newX - 1] === board[y][x] &&
+          !merged[newX - 1]
+        ) {
+          board[y][newX - 1] *= 2;
+          scoreVal += board[y][newX - 1];
+          board[y][x] = 0;
+          merged[newX - 1] = true;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-left--${moveDistance + 1}`);
+          }
+        } else if (newX !== x) {
+          board[y][newX] = board[y][x];
+          board[y][x] = 0;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-left--${moveDistance}`);
+          }
+        }
+      }
+    }
+  }
+
+  return hasChanged;
+}
+
+function moveRight() {
+  const originalBoard = board.map((row) => row.slice());
+  let hasChanged = false;
+
+  for (let y = 0; y < 4; y++) {
+    let merged = [false, false, false, false];
+
+    for (let x = 2; x >= 0; x--) {
+      if (board[y][x] !== 0) {
+        let newX = x;
+        let moveDistance = 0;
+
+        while (newX < 3 && board[y][newX + 1] === 0) {
+          newX++;
+          moveDistance++;
+        }
+
+        if (
+          newX < 3 &&
+          board[y][newX + 1] === board[y][x] &&
+          !merged[newX + 1]
+        ) {
+          board[y][newX + 1] *= 2;
+          scoreVal += board[y][newX + 1];
+          board[y][x] = 0;
+          merged[newX + 1] = true;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-right--${moveDistance + 1}`);
+          }
+        } else if (newX !== x) {
+          board[y][newX] = board[y][x];
+          board[y][x] = 0;
+          hasChanged = true;
+
+          const cellElement = rows[y].cells[x].firstChild;
+          if (cellElement) {
+            cellElement.classList.add(`move-right--${moveDistance}`);
+          }
+        }
+      }
+    }
+  }
+
+  return hasChanged;
```

Comment:
The 'moveRight' function, as with the other directional move functions, could benefit from a refactor to consolidate shared logic and simplify the codebase.